core/src/test/java/org/elasticsearch/node/internal/InternalSettingsPreparerTests.java

```diff
@@ -220,4 +222,19 @@ public String readText(String message, Object... args) {
         assertThat(settings.get("name"), is("prompted name 0"));
         assertThat(settings.get("node.name"), is("prompted name 0"));
     }
+
+    @Test
+    public void testPreserveSettingsClassloader() {
+        final ClassLoader classLoader = new URLClassLoader(new URL[] {});
```

Comment:
Can we use URLClassLoader.newInstance() instead? It doesn't require a createClassLoader permissions check i think, because the returned subclass is controlled and does proper security checks.